vertica_python/vertica/cursor.py

```diff
@@ -58,8 +58,7 @@ def execute(self, operation, parameters=None):
 
         self.rowcount = 0
         if self.last_execution:
-            # ToDo: can just empty the message buffer in connection if easy to do
-            self.connection.reset_connection()
```

Comment:
being able to leave an open connection seems like the correct behavior to me.
i think we should get rid of the connection reset (as in this pull request)

operations (not including iterate) should read all messages to conclusion, so they shouldn't cause a problem.

with iterate(), we can be left in a bad state if we try to execute another query mid iteration. But throwing an error in this case seems like it might be correct, since we don't want to silently flush the rest of the messages on the connection. Perhaps we could also explicitly add a flush() method that drops messages until we hit a ReadyForQuery, in case a client wants to purposely stop iteration for some reason.